### PR TITLE
Add a Quickstart section to the documentation

### PR DESCRIPTION
**Pull request by Ceasar**
The Quickstart shows how to create a simple Python web app which depends
on Redis, giving readers a sense of how Maestro works before proceeding
to the more in-depth sections explaining how to configure the
environment.

Also simplifies the introduction to make it less wordy.

@mpetazzoni

### CODE REVIEW

**Comment by Ceasar** [comment]
Would be worth adding an extended section on how to make Redis persistent too I think, just to show how the `volume` and `command` options are used.

File: docs/index.md [hunk 1]
```diff
@@ -1,53 +1,142 @@
...
+    image: redis
+    instances:
+      redis-1:
+        ship: localhost
+        ports:
+            redis: 6379
+        lifecycle:
+            running: [{type: tcp, port: redis}]
+  web:
+    image: ceasar/web
```
Comment (Ceasar) [inline docs/index.md]: This should be different. Not sure what value makes sense though.

File: docs/index.md [hunk 2]
```diff
@@ -1,53 +1,142 @@
...
+redis = Redis(host=redis_host, port=int(redis_port))
+
+@app.route('/')
+def hello():
+    redis.incr('hits')
+    return 'Hello World! I have been seen %s times.' % redis.get('hits')
+
+if __name__ == "__main__":
+    app.run(host="0.0.0.0", port=get_port('http'), debug=True)
+```
```
Comment (Ceasar) [inline docs/index.md]: Not sure if it's okay to copy the example from Fig so blatantly. Maybe we should make a slightly different example.

File: docs/index.md [hunk 3]
```diff
@@ -1,53 +1,142 @@
...
+Notice this makes use of two functions form
+[`maestro.guestutils`](guest-functions.md), `get_node_list()` and `get_port()`.
+`get_node_list()` provides a list of hostnames for the given service, and
+`get_port()` gets the port number for the given port name within an instance.
+
+Next we'll create our `Dockerfile` which tells Docker how to create an image
+that runs our Flask app.
+
+```
+FROM python:2.7
```
Comment (Ceasar) [inline docs/index.md]: When we worked together yesterday, we used a different image with `maestro` installed. Is that important? Or do we just need to add it to the requirements?

File: docs/index.md [hunk 4]
```diff
@@ -1,53 +1,142 @@
...
+ADD . /code
+WORKDIR /code
+RUN pip install -r requirements.txt
```
Comment (Ceasar) [inline docs/index.md]: Forgot to include the requirements file.

File: docs/index.md [hunk 5]
```diff
@@ -1,53 +1,142 @@
...
+   application. We want both services to run on `localhost`, and we want to
+   expose their default ports (6379 for Redis and 5000 to Flask) so that Flask
+   can reach Redis, and we can reach Flask. We also tell Maestro that `web`
+   depends on `redis`, so that Maestro doesn't start `web` until `redis` is
+   running. The `lifecycle` sections tell Maestro how to check if a service is
+   running successfully. In this case, our services are running correctly if
+   Maestro can reach them at their exposed ports.
+
+Next, we'll create our Flask app in ``app.py``, which simply print the number
+of times the index has been visited:
```
Comment (Ceasar) [inline docs/index.md]: print -> prints

**Comment by mpetazzoni** [comment]
Consider providing the required files under `docs/examples/<something>`.

I'm also actually trying to break off `index.md` into distinct documents/pages for ReadTheDocs, so I'd like to avoid adding content to it in favor of a new `quickstart.md` page.

**Comment by Ceasar** [comment]
Good ideas. I for splitting up the index, we should create a new issue splitting up the existing page, and then adjust this PR to just add a Quickstart page. I suppose we should create a new issue and discuss that there.

**Comment by mpetazzoni** [comment]
Thanks. I'll wait for the update on this one where you switch to a separate page for quickstart.

**Pull request closed without merging**